src/dtls13.c

```diff
@@ -2296,6 +2296,10 @@ int Dtls13RtxTimeout(WOLFSSL* ssl)
         return 0;
     }
 
+    /* Increase timeout on long timeout */
+    if (DtlsMsgPoolTimeout(ssl) != 0)
+        return -1;
```

Comment:
we error out once reached max timeout, so after about `max_timeout/DTLS_TIMEOUT_MULTIPLERS` numbers of retransmission. 

I think a better solution would be to explicitly set a maximum number of retransmissions than relying on the maximum timeout. I would just let the timeout saturate: stop its increase once it reaches the maximum. This means discarding the return value of `DtlsMsgPoolTimeout`. 

If you think that adding a explicitly counter of max retransmission is outside the scope for this PR leave it as it is for now so we have at least a limit that would stop hanging tests.

Reply:
I think that this change should be introduced separately, but I also don't see much advantage in changing this to a repeat counter instead of a timeout limit.

>I would just let the timeout saturate: stop its increase once it reaches the maximum.

We need a way to error out on a non-responsive peer. How would you suggest we error out? Or do you mean that we introduce a separate re-transmission counter from the timeout counter?

Reply:
Yes, I mean a separate re-transmission counter. After reaching MAX_NUMBER_OF_RTX we error out.

Reply:
That would mean that we should re-think our timeout setting API as currently we set the initial value and the maximum value. Let's tackle this outside of this PR.